pkg/api/handlers/compat/volumes.go

```diff
@@ -96,11 +96,17 @@ func CreateVolume(w http.ResponseWriter, r *http.Request) {
 		return
 	}
 
-	// See if the volume exists already
-	existingVolume, err := runtime.GetVolume(input.Name)
-	if err != nil && errors.Cause(err) != define.ErrNoSuchVolume {
-		utils.InternalServerError(w, err)
-		return
+	var (
+		existingVolume *libpod.Volume
+		err            error
+	)
+	if len(input.Name) != 0 || utils.IsLibpodRequest(r) {
```

Comment:
It's safe to drop the `IsLibpodRequest` bit, no reason this shouldn't work everywhere.

Reply:
Fixed.